pkg/controller/sparkapplication/controller.go

```diff
@@ -613,6 +614,9 @@ func (c *Controller) syncSparkApplication(key string) error {
 			return err
 		}
 	case v1beta2.CompletedState, v1beta2.FailedState:
+		if err := c.removeDriverPodFinalizer(app); err != nil {
+			return err
+		}
```

Comment:
I am wondering when you try to delete a  running SparkApplication, whether spark operator will need to wait the app to be completed or failed, and then it can remove the finalizer. If this is true, for a long running Spark job, like Spark streaming job, maybe it will never turned into completed or failed state and the finalizer will never be removed.

Reply:
You are right, finalizer is used for controling pod deletion, if the streaming driver pod is running, the owner spark job will never turn into completed or failed state, but if driver pod is deleted manually, or become status out of pod pending and pod running status, the streaming sparkapp will finally turn into completed or failed, at that time, driver finalizer will be deleted